server/api_server/nettools_endpoint.py

```
@@ -113,30 +118,48 @@ def traceroute(ip):
     try:
-        # Run speedtest-cli command using the resolved absolute path
+        # Detect if we're using the native Ookla binary or the python-based CLI
+        # Heuristic: if the executable is named exactly 'speedtest', it's likely the native one
+        is_native = os.path.basename(SPEEDTEST_CLI_PATH) == "speedtest"
+
+        if is_native:
+            cmd = [SPEEDTEST_CLI_PATH, "--format=json", "--accept-license", "--accept-gdpr"]
+        else:
+            cmd = [SPEEDTEST_CLI_PATH, "--secure", "--simple"]
+
         result = subprocess.run(
```

Ruff (0.14.10)

`subprocess` call: check for execution of untrusted input

(S603)